client/state/current-user/gravatar-status/actions.js

```diff
@@ -34,20 +41,40 @@ export function uploadGravatar( file, bearerToken, email ) {
 					dispatch( {
 						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
 					} );
+					dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_success' ) );
 				} );
 				fileReader.readAsDataURL( file );
 			} )
 			.catch( () => {
 				dispatch( {
 					type: GRAVATAR_UPLOAD_REQUEST_FAILURE
 				} );
+				dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) );
+				dispatch( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) );
```

Comment:
Since we're dispatching an action here, `GRAVATAR_UPLOAD_REQUEST_FAILURE` we can attach analytics as metadata to that action.

```jsx
const gravatarUploadFailure = flowRight(
	withAnalytics( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) ),
	withAnalytics( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) ),
	() => ( { type: 'GRAVATAR_UPLOAD_REQUEST_FAILURE' } )
)();
dispatch( gravatarUploadFailure );
```

see [client/state/analytics](https://github.com/Automattic/wp-calypso/blob/master/client/state/analytics/README.md) for more examples